janusgraph-cql/src/main/java/org/janusgraph/diskstorage/cql/CQLKeyColumnValueStore.java

```diff
@@ -143,16 +144,19 @@
      * @param tableName the name of the database table for storing the key/column/values
      * @param configuration data used in creating this store
      * @param closer callback used to clean up references to this store in the store manager
+     * @param shouldInitializeTable if true is provided the table gets initialized
      */
-    public CQLKeyColumnValueStore(final CQLStoreManager storeManager, final String tableName, final Configuration configuration, final Runnable closer) {
+    public CQLKeyColumnValueStore(final CQLStoreManager storeManager, final String tableName, final Configuration configuration, final Runnable closer, final Supplier<Boolean> shouldInitializeTable) {
```

Comment:
I'm curious why you chose to pass the `Supplier` from `CQLStoreManager` here. It seems like keeping the table existence checking logic contained in `CQLKeyColumnValueStore` would be consistent with how the keyspace is handled.

Reply:
I decided to do it that way for two reasons:
# The cluster containing the required metadata is private to the `CQLStoreManager` and can therefore not be accessed in `CQLKeyColumnValueStore`. Provide this information on the manager might be an option as well, but this would couple the manager an the store more together.
# Using `Supplier` instead of passing the information directly I chose because I did not want to  evaluate this for an open store. (`KeyColumnValueStore` already in `this.openStores`)

Reply:
Thanks for the explanation.